state/factory/blockpreparer_test.go

```diff
@@ -0,0 +1,107 @@
+package factory
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-core/v2/action"
+	"github.com/iotexproject/iotex-core/v2/blockchain/block"
+	"github.com/iotexproject/iotex-core/v2/test/identityset"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlockPreparer_PrepareOrWait(t *testing.T) {
+	preparer := newBlockPreparer()
+	prevHash := hash.Hash256b([]byte("previousHash"))
+	timestamp := time.Now()
+	mockBlk := &block.Block{}
+	called := false
+
+	// Mock mint function
+	mintFn := func() (*block.Block, error) {
+		if called {
```

Comment:
`called` kept being `false`, does the test plan to set it to `true` somewhere and check the error on L27 will be hit?